client_test/volume_test.py

```diff
@@ -0,0 +1,82 @@
+# Copyright Modal Labs 2023
+
+import pytest
+
+import modal
+from modal.exception import InvalidError
+from modal.volume import VolumeHandle
+
+from .supports.skip import skip_windows
+
+
+def dummy():
+    pass
+
+
+def test_volume_mount(client, servicer):
+    stub = modal.Stub()
+
+    dummy_modal = stub.function(
+        volumes={"/root/foo": modal.Volume()},
```

Comment:
I think @erikbern has mentioned him not liking the dict-style volume assigment of the current shared volumes and having some other idea of how to specify the mount point, and this might be a good time to introduce it since this is a new entity anyways.
Although I kind of like the "dict mounting" syntax :)

Reply:
I also don't love the dict-style - it's hard to extend in the future if you need to.

E.g. adding a `allow_cross_regional` flag per-shared volume, or adding a mount as `read-only` flag or such.

Reply:
Yeah it might be that it's better to be able to assign some metadata about the function-volume relation through some other notation, e.g.
```python
vol = Volume()

@stub.function(volumes=[vol.mount("/my_vol", read_only=True)])
```

Reply:
Should be pretty easy to add as a client-only "private" type `AttachedVolume` which is returned by `vol.mount` and is the required type for the `volumes` parameter to functions (but the underlying Modal object type would still be Volume)

Reply:
Let's leave API improvements for a follow-up commit, if you're okay with that? We have time to change things around before this features goes public.

There's also the idea that I floated of unifying volume mounting for all different kinds of volumes, e.g.:

```python
vol1 = modal.SharedVolume()
vol2 = modal.NewFangledVolume()

@stub.function(volumes={"/vol1": vol1, "/vol2": vol2}) # <-- Shared volumes and "volumes" are attached to the function the same way
def f():
    ...
```

Reply:
👍